### PR TITLE
MRG: Use h5io and fix tests

### LINKED ISSUE #2420: Spinning off the _hdf5 module?
There was just a discussion on the scikit learn listserv where the _hdf5 module in MNE came up. I feel like we've chatted about this here and there in the past, but wanted to open it up more formally.

That _hdf5 module is pretty darn useful, and I think it's too bad that it currently exists hidden to the user. Does anyone think it would be better if that was either a module that you could find more easily (e.g., mne.io.hdf5), or even spun off into its own mini package or something?

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #2420.

I also included some fixes I needed to get the test suite to run on my machine without as many warnings.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
CIs are happy, ready for review/merge from my end

File: mne/viz/epochs.py [hunk 1]
```diff
@@ -371,6 +371,47 @@ def plot_epochs_trellis(epochs, epoch_idx=None, picks=None, scalings=None,
     fig : Instance of matplotlib.figure.Figure
         The figure.
     """
+    _plot_epochs_trellis(epochs, epoch_idx, picks, scalings, title_str,
+                         show, block, n_epochs)
```
Comment (Eric89GXL) [inline mne/viz/epochs.py]: this fixes a problem where `plot_epochs(..., trellis=True)` was throwing a deprecationwarning when only `plot_epochs_trellis(...)` should have been, cc @jaeilepp

**Comment by agramfort** [comment]
I would commit the entire h5io folder with tests

**Comment by Eric89GXL** [comment]
Really? Why bother? We also don't do it with any of the other `external` deps.

**Comment by dengemann** [comment]
> Really? Why bother? We also don't do it with any of the other external deps.

Wouldn't it be easier to maintain if we just had to copy and paste the entire package once it is updated upstream?

**Comment by Eric89GXL** [comment]
I'd rather copy-paste a single file than a bunch of files

**Comment by agramfort** [comment]
See how it's done with joblib in sklearn

**Comment by Eric89GXL** [comment]
Looks like they don't copy the test folder, either. Compare:

https://github.com/scikit-learn/scikit-learn/tree/master/sklearn/externals/joblib
https://github.com/joblib/joblib/tree/master/joblib

I assume they copied everything else because they needed it to function. Maybe you didn't notice, but I actually have h5io structured so that `_h5io.py` is the only file with useful code in it, with no other files needed for it to work. So any updates to `h5io` will be in that single file, at least for now. This structuring was intentional to make it easy to include in `externals` in various packages actually...

**Comment by dengemann** [comment]
Assuming that the package in question fullfils its testing duty, this makes sense, I have to admit. We don't want to run these tests.

**Comment by agramfort** [comment]
of the tests yet I would put the file in 

mne/externals/h5io/_h5io.py

so the import from mne is mne.externals.h5io

thanks @Eric89GXL

**Comment by Eric89GXL** [comment]
@agramfort done

**Pull request merged**